File: LeetCode/Medium/0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        ListNode node = new ListNode();
        ListNode curr = node;

        while (head != null) {
            boolean flag = false;
            int a = head.val;
            int b = 0;
            head = head.next;

            if (head != null) {
                b = head.val;
                head = head.next;
                flag = true;
            }

            if (flag) {
                curr.next = new ListNode(b);
                curr = curr.next;
                curr.next = new ListNode(a);
                curr = curr.next;
            } else {
                curr.next = new ListNode(a);
                curr = curr.next;
            }
        }

        return node.next;
    }
}

File: LeetCode/Medium/0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun swapPairs(head: ListNode?): ListNode? {
        val node = ListNode(0)
        var curr = node
        var h = head

        while (h != null) {
            if (h.next != null) {
                curr.next = ListNode(h.next.`val`)
                curr.next.next = ListNode(h.`val`)
                curr = curr.next.next
                h = h.next.next
            } else {
                curr.next = ListNode(h.`val`)
                curr = curr.next
                h = h.next
            }
        }

        return node.next
    }
}